features/steps/component.go

```diff
@@ -87,11 +90,17 @@ func (c *Component) InitAPIFeature() *componentTest.APIFeature {
 	return c.APIFeature
 }
 
-// Reset resets the search controller component
-func (c *Component) Reset() *Component {
+// Reset resets the search controller component to its default values
+func (c *Component) Reset() error {
+	if c.cfg == nil {
```

Comment:
So not sure the Reset method is being used as a reset?

It is being used to set the configuration if it is nil, that means nothing is being reset.

Reply:
Think about creating a new method if you want this functionality,`func SetConfig() error`

Im still unsure (at this moment in peer review) why you need this functionality

Reply:
I have addressed this in the big comment below